code/modules/reagents/reagent_containers/food/drinks/yoke.dm

```diff
@@ -0,0 +1,115 @@
+	var/list/can_positions = list( // these are the correct positions for energy drinks achieved via trial and error
+		list(4, -4),
+		list(-2, -2),
+		list(-8, 0),
+		list(9, -3),
+		list(4, 0),
+		list(-2, 3)
+	)
+
+/obj/item/storage/box/fancy/yoke/fill()
+	. = ..()
+	for(var/obj/item/reagent_containers/food/drinks/cans/C in contents)
+		cans += C
+
+/obj/item/storage/box/fancy/yoke/update_icon()
+	for(var/thing in underlays)
+		underlays -= thing
+
+	for(var/i = 1 to length(cans))
+		var/mutable_appearance/can = mutable_appearance(cans[i].icon, cans[i].icon_state)
+		var/list/positions = can_positions[i]
+		can.pixel_x = positions[1]
+		can.pixel_y = positions[2]
+		underlays += can
+
+/obj/item/storage/box/fancy/yoke/MouseDrop(mob/user)
+	if(user != usr || !ishuman(user) || use_check_and_message(user) || user.get_active_hand())
+		return
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
+	if(H.hand)
+		temp = H.organs_by_name[BP_L_HAND]
+	if(temp && !temp.is_usable())
+		to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
+		return
+
+	to_chat(user, SPAN_NOTICE("You pick up \the [src]."))
+	user.put_in_hands(src)
+
+/obj/item/storage/box/fancy/yoke/attack_hand(mob/user)
+	if(!length(contents)) // no more cans, continue as normal
+		return ..()
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
+		if(H.hand)
+			temp = H.organs_by_name[BP_L_HAND]
+		if(temp && !temp.is_usable())
+			to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
+			return
+	
+	var/obj/item/reagent_containers/food/drinks/cans/C = cans[length(cans)]
+	cans -= C
+	remove_from_storage(C, get_turf(user))
+	user.put_in_hands(C)
+	update_icon()
+
+/obj/item/storage/box/fancy/yoke/attackby(obj/item/W, mob/user)
+	to_chat(user, SPAN_WARNING("\The [src] cannot be refilled with items!"))
+
+/obj/item/storage/box/fancy/yoke/zorasoda
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zorasoda = 6)
+
+/obj/item/storage/box/fancy/yoke/zorakois
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zorakois = 6)
+
+/obj/item/storage/box/fancy/yoke/zoraphoron
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoraphoron = 6)
+
+/obj/item/storage/box/fancy/yoke/zorahozm
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zorahozm = 6)
+
+/obj/item/storage/box/fancy/yoke/zoravenom
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoravenom = 6)
+
+/obj/item/storage/box/fancy/yoke/zoraklax
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoraklax = 6)
+
+/obj/item/storage/box/fancy/yoke/zoracthur
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoracthur = 6)
+
+/obj/item/storage/box/fancy/yoke/random
+	starts_with = list()
+
+/obj/item/storage/box/fancy/yoke/random/fill()
+	for(var/i = 1; i <= 6, i++)
```

Comment:
for the future, these loops are slower than using byond's (aka `for(var/i = 1 to 6)`